### PR TITLE
Materials Project data loaders

### PR DESCRIPTION
**Pull request by ncfrey**
This PR adds two new loader functions to MolNet.

### Summary
- `load_mp_metallicity` loads a dataset of inorganic crystal structures from the Materials Project labeled as metals or nonmetals.
- `load_mp_formation_energy` loads a dataset of inorganic crystal structures from the Materials Project with calculated formation energies.
- All four datasets from `molnet.load_function.materials_datasets` are now uploaded to the DeepChem AWS bucket.

I also fixed a bug in the `SineCoulombMatrix` featurizer

CC @nd-02110114

### CODE REVIEW

File: deepchem/molnet/load_function/material_datasets/load_mp_formation_energy.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+from deepchem.trans import Transformer
+from deepchem.splits.splitters import Splitter
+from deepchem.molnet.defaults import get_defaults
+
+from typing import List, Tuple, Dict, Optional, Union, Any, Type
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_DIR = deepchem.utils.get_data_dir()
+MPFORME_URL = 'http://deepchem.io.s3-website-us-west-1.amazonaws.com/datasets/mp_formation_energy.tar.gz'
```
Comment (nd-02110114) [inline deepchem/molnet/load_function/material_datasets/load_mp_formation_energy.py:17]: The URL is not correct. (There is a same problem in `load_mp_metallicity.py`)

```python
MPFORME_URL = 'https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/mp_formation_energy.tar.gz'
```

File: deepchem/molnet/load_function/material_datasets/load_mp_formation_energy.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  # Load .tar.gz file
+  if featurizer.__class__.__name__ in supported_featurizers:
+    dataset_file = os.path.join(data_dir, 'mp_formation_energy.tar.gz')
+    deepchem.utils.untargz_file(dataset_file, dest_dir=data_dir)
+    dataset_file = os.path.join(data_dir, 'mp_formation_energy.json')
+
+    if not os.path.exists(dataset_file):
+      deepchem.utils.download_url(url=MPFORME_URL, dest_dir=data_dir)
+      deepchem.utils.untargz_file(
+          os.path.join(data_dir, 'mp_formation_energy.tar.gz'), data_dir)
```
Comment (nd-02110114) [inline deepchem/molnet/load_function/material_datasets/load_mp_formation_energy.py:173]: I seem these lines doesn't work... (see #2083)
Please check whether to work except pytest like this.

```bash
$ python
>>> import deepchem as dc
>>> tasks, datasets, transformers = dc.molnet.load_mp_formation_energy(reload=False)
```

**Comment by ncfrey** [comment]
I made the fixes pointed out by @nd-02110114 and resolved merge conflicts! I think this is ready to merge.

**Comment by rbharath** [comment]
Looks good on my end! I'll wait to see if @nd-02110114 has any more comments and will plan to merge tomorrow morning otherwise :)

**Comment by nd-02110114** [comment]
LGTM

**Pull request merged**